service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java

```diff
@@ -245,4 +245,47 @@ public List<GenericAssayDataCountItem> fetchGenericAssayDataCounts(List<String>
             })
             .collect(Collectors.toList());
     }
+    
+    @Override
+    public List<GenePanelCountItem> fetchGenePanelCounts(List<String> sampleIds, List<String> studyIds) {
+        // sample ID, molecular profile ID pairs
+        List<MolecularProfileCaseIdentifier> profileIds = molecularProfileService.getMolecularProfileCaseIdentifiers(studyIds, sampleIds);
+        return genePanelService.fetchGenePanelDataInMultipleMolecularProfiles(profileIds)
+            .stream()
+            .collect(Collectors.groupingBy(GenePanelData::getMolecularProfileId))
+            .entrySet().stream()
+            .map(this::toGenePanelCountItem)
+            .collect(Collectors.toList());
+    }
+
+    private GenePanelCountItem toGenePanelCountItem(Map.Entry<String, List<GenePanelData>> entry) {
+        GenePanelCountItem count = new GenePanelCountItem();
+        count.setMolecularProfileId(entry.getKey());
+        List<ClinicalDataCount> counts = entry.getValue().stream()
+            .map(this::fixWholeExomeMadness)
+            .collect(Collectors.groupingBy(GenePanelData::getGenePanelId))
+            .entrySet().stream()
+            .map(genePanelsOfId -> toGenePaneCount(genePanelsOfId, entry.getKey()))
+            .collect(Collectors.toList());
+        count.setCounts(counts);
+        return count;
+    }
+    
+    private GenePanelData fixWholeExomeMadness(GenePanelData coolNulls) {
+        if (coolNulls.getGenePanelId() == null) {
+            coolNulls.setGenePanelId("Whole exome seq.");
```

Comment:
If this text also being used elsewhere, we should associate this text with a parameter.